### PR TITLE
feat: support PWA

### PR DESCRIPTION
**Pull request by SiongSng**
# Description

support PWA

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -24,4 +24,4 @@ dist-ssr
 *.ntvs*
 *.njsproj
 *.sln
-*.sw?
+*.sw?
```
Comment (banahaker) [inline .gitignore]: 蛤

**Review by banahaker — APPROVED** [review]
not bad

**Review by banahaker — CHANGES_REQUESTED** [review]
Why delete the `plugin/index.ts` file

**Comment by SiongSng** [comment]
Because it has been merged into vite.config.ts

**Comment by banahaker** [comment]
ok

**Review by banahaker — APPROVED** [review]

**Pull request merged**